### PR TITLE
[Mobile Payments] Payments button on hub menu

### LINKED ISSUE #7358: Add "payments" button that opens to the IPP hub screen from the more menu screen
This can be done via added "Payments" button to the more menu screen:

Image

(Android design, it should give us an idea of how to do it on iOS)

(make sure that it works on small screen, and in a dark mode well)

### PR DESCRIPTION
**Pull request by toupper**
Closes: #7358 

### Description

With this PR we add the payments button in the Hub Menu, linking the right action to it, so it can open the IPP screen when tapped.

#### 

1. Go to Menu
2. The Payments button should be displayed the first
3. Tap on it. It should open the IPP Settings Screen

Please also check that when the feature flag is disabled the button is now shown.

### Screenshots

---
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Hub Menu/HubMenuViewModel.swift [hunk 1]
```diff
@@ -81,6 +81,10 @@ final class HubMenuViewModel: ObservableObject {
     func setupMenuElements() {
         menuElements = [.woocommerceAdmin, .viewStore, .reviews]
 
+        if featureFlagService.isFeatureFlagEnabled(.paymentsHubMenuSection) {
+            menuElements.insert(.payments, at: 0)
+        }
+
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Hub Menu/HubMenuViewModel.swift:87]: Clean and nice 💯

File: WooCommerce/WooCommerceTests/ViewRelated/HubMenu/HubMenuViewModelTests.swift [hunk 2]
```diff
@@ -29,6 +29,17 @@ final class HubMenuViewModelTests: XCTestCase {
         XCTAssertFalse(viewModel.menuElements.contains(.inbox))
     }
 
+    func test_menuElements_do_not_include_payments_when_feature_flag_is_off() {
```
Comment (iamgabrielma) [inline WooCommerce/WooCommerceTests/ViewRelated/HubMenu/HubMenuViewModelTests.swift:32]: Should we go the other way around with this test, so we do not need to remove it once this feature goes live? Something like `test_menuElements_include_payments_when_feature_flag_is_on` instead?

File: WooCommerce/Classes/ViewRelated/Hub Menu/HubMenuViewModel.swift [hunk 3]
```diff
@@ -204,6 +217,8 @@ extension HubMenuViewModel {
     }
 
     private enum Localization {
+        static let payments = NSLocalizedString("Payments",
+                                                comment: "Title of one of the hub menu options")
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Hub Menu/HubMenuViewModel.swift]: Should we change this comment to be more descriptive for the translators? Something like:

```
static let payments = NSLocalizedString("Payments", comment: "Title of the Payments menu in the hub menu")
```

**Review by iamgabrielma — APPROVED** [review]
LFGTM 🚀  works nicely and switching the feature flag off also disables the feature, as expected.

Like in the previous PR I see 2 tests have failed in BuildKite, `test_setFeatureAnnouncementDismissed_with_remindLater_true_stores_reminder_date_in_two_weeks `  (expected while testing) and `test_loadHubMenuTabBadge_when_both_badges_should_be_shown_calls_onMenuBadgeShouldBeDisplayed_with_type_primary`, which seems to time-out when ran in the test suite along others. Something to keep an eye on in other related PRs as well 👍

File: WooCommerce/Classes/ViewRelated/Hub Menu/HubMenuViewModel.swift (hunk 3 above)
Comment (toupper) [reply WooCommerce/Classes/ViewRelated/Hub Menu/HubMenuViewModel.swift]: Yeah! Done in 59baeebcfa38be9b6ee4d43b92cd91c0d9e19187

File: WooCommerce/WooCommerceTests/ViewRelated/HubMenu/HubMenuViewModelTests.swift (hunk 2 above)
Comment (toupper) [reply WooCommerce/WooCommerceTests/ViewRelated/HubMenu/HubMenuViewModelTests.swift:32]: Actually I think both cases are important, when the feature flag is on and off. I added the case you mentioned in 1320c6331219eb23329d465e60ab228fb142abe2

**Pull request merged**